### ISSUE TITLE
Add comments on templates to show where the HTML is coming from

### ISSUE THREAD

**Issue by ralsina** [getnikola/nikola#2985]
For example, it happens that you are trying to edit a theme's index page.

You look at it and ... which template is generating that?

We could add comments in all templates and in blocks so that is explicit.

Sure, it makes the output larger, but if you are minifying html, the comments go away.

**Comment by Kwpolska** [comment]
Mako and Jinja have a comment feature. Those do not end up in the source output. We already use them in some places (in particular to specify the file encoding for Mako)

**Comment by ralsina** [comment]
Yeah, but I *do* want them in the output so people trying to make a theme know what template to tweak

**Comment by Kwpolska** [comment]
Then it gets a huge :-1: from me. This is completely irrelevant to most people and will look strange to people reading the source, unaware of what it means.

**Comment by michaelb42** [comment]
In combination with a "draft" and "release" switch this would become really useful while editing themes, templates and changing styles. Applying some kind of "build configuration" to the filters, etc. one could easily choose between different tidy profiles and remove those comments in production builds, generate non-minified or minified css, etc. :+1:

**Comment by michaelb42** [comment]
Hm, just recognized that nikola supports a --conf= option (is it documented somewhere?) that could be used to differentiate between a "draft" and a "release".

**Comment by ralsina** [comment]
Could be done as an option enabled/disabled by a switch.

Honestly, trying to do a theme without this is complicated. And I wrote the damned theme engine.

**Comment by aknrdureegaesr** [comment]
https://github.com/getnikola/nikola/pull/3760 will at least offer logging which templates went into which output file (when that has been merged finally, might happen any month now). This is not _exactly_ what was wanted here, but a different approach to roughly the same problem field.
